test/e2e/framework/endpoints/wait.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+This soak tests places a specified number of pods on each node and then
+repeatedly sends queries to a service running on these pods via
+a serivce
+*/
+
+package endpoints
+
+import (
+	"fmt"
+	"time"
+
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+// WaitForEndpoint waits for the specified endpoint to be ready.
+func WaitForEndpoint(c clientset.Interface, ns, name string) error {
+	for t := time.Now(); time.Since(t) < framework.EndpointRegisterTimeout; time.Sleep(framework.Poll) {
```

Comment:
Should we also move EndpointRegisterTimeout in here?

Reply:
Yeah, I moved `EndpointRegisterTimeout ` to this file and made it private since no one else uses this constant.